src/graphql/graphql-api.php

```diff
@@ -0,0 +1,255 @@
+<?php
+/**
+ * GraphQL API
+ *
+ * @package vip-block-data-api
+ */
+
+namespace WPCOMVIP\BlockDataApi;
+
+use GraphQLRelay\Relay;
+
+defined( 'ABSPATH' ) || die();
+
+/**
+ * GraphQL API to offer an alternative to the REST API.
+ */
+class GraphQLApi {
+	/**
+	 * Initiatilize the graphQL API by hooking into the graphql_register_types action, 
+	 * which only fires if WPGraphQL is installed and enabled, and is further controlled 
+	 * by the vip_block_data_api__is_graphql_enabled filter.
+	 */
+	public static function init() {
+		add_action( 'graphql_register_types', [ __CLASS__, 'register_types' ] );
+	}
+
+	/**
+	 * Extract the blocks data for a post, and return back in the format expected by the graphQL API.
+	 *
+	 * @param  \WPGraphQL\Model\Post $post_model Post model for post.
+	 * 
+	 * @return array
+	 */
+	public static function get_blocks_data( $post_model ) {
+		$post_id = $post_model->ID;
+		$post    = get_post( $post_id );
+
+		$content_parser = new ContentParser();
+
+		$parser_results = $content_parser->parse( $post->post_content, $post_id );
+
+		// We need to not return a WP_Error object, and so a regular exception is returned.
+		if ( is_wp_error( $parser_results ) ) {
+			Analytics::record_error( $parser_results );
+
+			// Return API-safe error with extra data (e.g. stack trace) removed.
+			return new \Exception( $parser_results->get_error_message() );
+		}
+
+		$parser_results['blocks'] = array_map(function ( $block ) {
```

Comment:
is there no phpcs on this repo?
```suggestion
		$parser_results['blocks'] = array_map( function ( $block ) {
```

Reply:
Huh, weird! We do have `phpcs` configured, and it seems to spot other problems in this file just fine, but it's not enforcing parentheses spacing. Our [phpcs configuration](https://github.com/Automattic/vip-go-mu-plugins/blob/develop/phpcs.xml.dist) is a close relation to [vip-go-mu-plugins's config](https://github.com/Automattic/vip-go-mu-plugins/blob/develop/phpcs.xml.dist) which appears to be roughly the same. Looking into what sniff might be missing here.

Reply:
Figured it out. I noticed if I add a missing space a couple of lines up, PHPCS will report it:

```php
$block['id'] = Relay::toGlobalId('ID', wp_unique_id() );
//                               ^ missing space
```

```bash
FILE: src/graphql/graphql-api.php
--------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
--------------------------------------------------------------------------
 73 | ERROR | [x] Expected 1 spaces after opening parenthesis; 0 found
    |       |     (PEAR.Functions.FunctionCallSignature.SpaceAfterOpenBracket)
```

However, that same `PEAR.Functions.FunctionCallSignature.SpaceAfterOpenBracket` sniff doesn't kick in for `array_map()`. After looking at our configuration, I noticed we have [some sniffs disabled](https://github.com/Automattic/vip-go-mu-plugins/blob/develop/phpcs.xml.dist#L73-L76) related to anonymous functions and function calls. If I remove those, we'll now get multiple errors for the multi-line function call:

```
 72 | ERROR | [x] Opening parenthesis of a multi-line function call must be the last content on the line
    |       |     (PEAR.Functions.FunctionCallSignature.ContentAfterOpenBracket)
 74 | ERROR | [x] Only one argument is allowed per line in a multi-line function call (PEAR.Functions.FunctionCallSignature.MultipleArguments)
 74 | ERROR | [x] Closing parenthesis of a multi-line function call must be on a line by itself (PEAR.Functions.FunctionCallSignature.CloseBracketLine)
```

So, in short, if we want to disable those sniffs to use some standard PHP `array_map( function( ... ) {` practices, we also lose the sniff that tells us about weird spacing after `array_map()`.

Fixed manually.